server/controller/src/main/java/ai/starwhale/mlops/schedule/impl/k8s/reporting/PodEventHandler.java

```diff
@@ -73,19 +73,18 @@ private void reportRunStatus(V1Pod pod) {
             return;
         }
 
-        // https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination
-        if (pod.getMetadata() != null && pod.getMetadata().getDeletionTimestamp() != null) {
-            log.info("pod {} is being deleted", pod.getMetadata().getName());
-            return;
-        }
         Long rid = getJobNameAsId(pod);
         if (rid == null) {
             return;
         }
 
         RunStatus runStatus;
         var phase = pod.getStatus().getPhase();
-        if (StringUtils.hasText(phase)) {
+        // https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination
+        if (pod.getMetadata() != null && pod.getMetadata().getDeletionTimestamp() != null) {
+            log.info("pod {} is being deleted", pod.getMetadata().getName());
+            runStatus = RunStatus.FAILED;
```

Comment:
Will this operation turn a successful run into a failed one?

Reply:
> Will this operation turn a successful run into a failed one?

No, it is not blocked by other modules. 
The logic here is not accurate, though. And, I have modified it.